src/hooks/useMaciPoll.ts

```diff
@@ -0,0 +1,53 @@
+import { getPoll, type Signer, type IGetPollData } from "maci-cli/sdk";
+import { useEffect, useMemo, useState } from "react";
+
+import { config } from "~/config";
+import { useEthersSigner } from "./useEthersSigner";
+
+export interface IUseMaciPollData {
+  isLoading: boolean;
+  votingEndsAt: Date;
+  pollData?: IGetPollData;
+}
+
+export const useMaciPoll = (): IUseMaciPollData => {
+  const signer = useEthersSigner();
+  const [isLoading, setIsLoading] = useState(false);
+  const [pollData, setPollData] = useState<IGetPollData>();
+
+  const votingEndsAt = useMemo(
+    () =>
+      pollData
+        ? new Date(
+            Number(pollData.deployTime) * 1000 +
+              Number(pollData.duration) * 1000,
+          )
+        : new Date(),
```

Comment:
So if no poll data, we just set this to the current timestamp?

Reply:
I'm ok with happy path for now, but maybe we add a comment that we should return an error if we can't get `pollData`? Or maybe this is a bigger issue than just this function if we're unable to fetch `pollData`

Reply:
If there is no poll data, poll information is not shown.